src/data-access/repositories/test/mongo-part-repository.spec.ts

```diff
@@ -262,20 +263,118 @@ describe(`${MongoPartRepository.name}`, () => {
     })
   })
 
+  describe(`${MongoPartRepository.prototype.getParts.name}`, () => {
+    it('gets zero parts from database when no parts for given segmentId exist', async () => {
+      const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+      const mongoParts: MongoPart[] = [createMongoPart({segmentId: 'someSegmentId'})]
+      const nonExistingId: string = 'nonExistingId'
+      const db: Db = testDatabase.getDatabase()
+      await testDatabase.populateDatabaseWithParts(mongoParts)
+
+      when(mongoConverter.convertParts(anything())).thenReturn([])
+      const testee: PartRepository = createTestee({
+        mongoConverter: mongoConverter,
+      })
+
+      await testee.getParts(nonExistingId)
+
+      const [capturedMongoParts] =  capture(mongoConverter.convertParts).first()
+      expect(capturedMongoParts.length).toBe(0)
+      expect((await db.collection(COLLECTION_NAME).find().toArray()).length).toBe(1)
+    })
```

Comment:
As quickly mention before lunch, by changing the test to the suggested, including the change to `createTestee()`, the test should never fail, as the expect is checking the length of the mocked return value.

Propably need a longer chat about this one.

Reply:
Your implementation fails if we change the implementation, but not the functionality. I.e. if we decided to no longer use the MongoEntityConverter and just instantiate the entities directly, then this test would fail even though it still returns zero entities on a non-existing id.

And as long as the correct functionality is there, the test shouldn't fail.

Reply:
Since my last comment, i have also come to a similar conclusion.

I will be updating the 3 test wiith the signature as the one commented on, to the suggested signature.
Will also add a comment stating that while we have and mock the `MongoEntityConverter`, the tests should never fail.